File: scripts/build-standalone.mjs
import fs from "fs";
import path from "path";
import { fileURLToPath } from "url";

const __dirname = path.dirname(fileURLToPath(import.meta.url));
const root = path.join(__dirname, "..");

const css = fs.readFileSync(path.join(root, "css", "styles.css"), "utf8");
const scenario = fs.readFileSync(path.join(root, "js", "scenario.js"), "utf8");
const game = fs.readFileSync(path.join(root, "js", "game.js"), "utf8");
let index = fs.readFileSync(path.join(root, "index.html"), "utf8");

const scenarioInline = scenario
  .replace(/export const scenarioMeta = \{[\s\S]*?\};\s*\n?/, "")
  .replace("export const beats", "const beats");

const gameInline = game.replace(/^import \{ beats \} from "\.\/scenario\.js";\s*\r?\n/m, "");

const script = `${scenarioInline.trim()}\n\n${gameInline.trim()}`;

index = index.replace(
  /<link rel="stylesheet" href="css\/styles\.css" \/>/,
  `<style>\n${css}\n</style>`
);
index = index.replace(
  /<script type="module" src="js\/game\.js"><\/script>/,
  `<script>\n${script}\n</script>`
);

const out = path.join(root, "prompt-orchestrator-standalone.html");
fs.writeFileSync(out, index, "utf8");
console.log("Wrote", out);

File: scripts/build-standalone.py
import re
from pathlib import Path

root = Path(__file__).resolve().parent.parent
css = (root / "css" / "styles.css").read_text(encoding="utf-8")
scenario = (root / "js" / "scenario.js").read_text(encoding="utf-8")
game = (root / "js" / "game.js").read_text(encoding="utf-8")
index = (root / "index.html").read_text(encoding="utf-8")

scenario_inline = re.sub(
    r"export const scenarioMeta = \{[\s\S]*?\};\s*\n?",
    "",
    scenario,
    count=1,
)
scenario_inline = scenario_inline.replace("export const beats", "const beats", 1)

game_inline = re.sub(
    r'^import \{ beats \} from "\./scenario\.js";\s*\r?\n',
    "",
    game,
    count=1,
    flags=re.MULTILINE,
)

script = scenario_inline.strip() + "\n\n" + game_inline.strip()

index = index.replace(
    '<link rel="stylesheet" href="css/styles.css" />',
    f"<style>\n{css}\n</style>",
)
index = index.replace(
    '<script type="module" src="js/game.js"></script>',
    f"<script>\n{script}\n</script>",
)

out = root / "prompt-orchestrator-standalone.html"
out.write_text(index, encoding="utf-8")
print("Wrote", out)
